typescript/sdk/src/hook/RadixHookReader.ts

```diff
@@ -0,0 +1,107 @@
+import { RadixSDK, RadixSigningSDK } from '@hyperlane-xyz/radix-sdk';
+import { Address, WithAddress, assert, rootLogger } from '@hyperlane-xyz/utils';
+
+import { ChainMetadataManager } from '../metadata/ChainMetadataManager.js';
+
+import {
+  DerivedHookConfig,
+  HookType,
+  IgpHookConfig,
+  MerkleTreeHookConfig,
+} from './types.js';
+
+export class RadixHookReader {
+  protected readonly logger = rootLogger.child({
+    module: 'RadixHookReader',
+  });
+
+  constructor(
+    protected readonly metadataManager: ChainMetadataManager,
+    protected readonly sdk: RadixSDK | RadixSigningSDK,
+  ) {}
+
```

Comment:
nit: Can you implement a function that returns the hook type directly? 

```typecript
async deriveHookType(address: Address): Promise<HookType>{
   ....
}
```

so you can then reuse it in the `deriveHookConfig` instead of having to call multiple functions here?